app/views/users/_non_friend.html.erb

```diff
@@ -0,0 +1,39 @@
+<section class="container">
+  <% if non_friend?(@user) %>
+    <div class="card">
+      <div class="card-body">
+        <p class="card-title ">Name: <span class="text-warning user_name"><%= @user.name %></span></p>
+        <div class="">
+          <h5 class="recent_posts">Recent Posts: </h5>
+          <% @posts.each do |post| %>
+            <div class="card mb-4">
+              <div class="card-body">
+                  <p class="text-italic"><%= post.content %></p>
+              </div>
+              <div class="card-footer">
+                <p>Creation Date: <span class="italic"><%= time_ago_in_words(post.created_at) %></span></p>
+                <p>Post Info: <%= link_to 'View Post', root_path , class: 'btn btn-sm btn-success' %> </p>
+              </div>
+            </div>
+          <% end %>
+        </div>
+
+        <div class="mutual-friends">
+          <h4 class="text-italic">Mutual Friends: </h4>
+          <% @mutual_friends.each do |friend|%>
+            <p><%= friend.name %></p>
+          <% end %>
+        </div>
+      </div>
+      
+      <div class="card-footer">
+        <button class="btn btn-sm btn-success">
+          <%= link_to 'Invite to Friendship', send_request_path(@user.id), class: 'text-light' %>
+        </button>
+        <button class="btn btn-sm btn-dark">
+          <%= link_to 'Back', users_path, class: 'text-light'%>
+        </button>
+      </div>
+    </div>
+  <% end %>
```

Comment:
Kindly remove that the logic. Move that to the view helpers.

Reply:
Thank you @maelfosso for the timely go ahead and the note of the changes required before merging.